great_expectations/core/expectation_suite.py

```diff
@@ -591,12 +594,8 @@ def render(self) -> None:
 
     def identifier_bundle(self) -> _IdentifierBundle:
         # Utilized as a custom json_encoder
-        from great_expectations.data_context.data_context.context_factory import project_manager
-
         if not self.id:
-            expectation_store = project_manager.get_expectations_store()
-            key = expectation_store.get_key(name=self.name, id=None)
-            expectation_store.add(key=key, value=self)
+            raise ExpectationSuiteNotAddedToStoreError(name=self.name)
```

Comment:
We want to fail at this point otherwise we will JSON encode identifier bundles with null ids (making it impossible to retrieve the object later)